src/mol-model/structure/structure/util/lookup3d.ts

```diff
@@ -86,6 +86,53 @@ export class StructureLookup3D {
         return ctx.result;
     }
 
+    nearest(x: number, y: number, z: number): { index: number, unit: Unit, squaredDistance: number } | undefined {
+        return this._nearest(x, y, z);
+    }
+
+    _nearest(x: number, y: number, z: number): { index: number, unit: Unit, squaredDistance: number } | undefined {
+        const ctx = this.findContext;
+        const { units, elementCount } = this.structure;
+
+        if (!elementCount) return undefined;
+
+        const radiusIncrement = 0.1; // how to choose a good increment?
+        const startingRadius = this.distanceTo(x, y, z);
+        let radius = startingRadius < 0 ? radiusIncrement : startingRadius + radiusIncrement;
+
+        while (true) {
+            const closeUnits = this.unitLookup.find(x, y, z, radius, ctx.closeUnitsResult);
+            radius += radiusIncrement;
```

Comment:
I am not sure this is a good approach which will work in general. Maybe a better one would be to sort the units based on distance to the query point and then query the units in that order, keeping the distance to the closest point. Then once the distance starts to be greater than the current closest points, you can stop the iteration.